### ISSUE TITLE
Provide A Flatpak Bundle of Syanpse

### ISSUE THREAD

**Issue by Saroufim** [MichaelAquilina/synapse-project#16]
It would be nice to have Synapse as a flatpak bundle to install on all distributions.

**Comment by MichaelAquilina** [comment]
Thanks for the request :) I dont have any experience building flatpak bundles, and very little experience using flatpak. I will try give this a look when I have the time but in the meanwhile may I suggest you also take a look at trying it out yourself ;)

**Comment by gnacho** [comment]
+1 to this. I'm unable to use it as is not available for Arch.

**Comment by MichaelAquilina** [comment]
Unfortunately, building a flatpak looks quite complicated (just looked at some manifest examples for inspiration). Will try give it a go again when reading the docs but help is very much appreciated if someone can give a hand :)

**Comment by gnacho** [comment]
I wish I could but I'm not a developer, just a user :(

**Comment by smac89** [comment]
What about a snap package? Is there anything more we gain from flatpak that cannot be handled by snaps?

I got started on one but I am running into compiler errors for some reason...

```yaml
name: synapse-project
# title: Synapse Project
# license: 'GPL2,LGPL2.1'
version: 0.4.6
summary: Powerful fuzzy search launcher for a multitude of tasks
description: >
  Synapse is a semantic launcher written in Vala that you can use to start 
  applications as well as find and access relevant documents and files by making use of the Zeitgeist engine.
architectures: [amd64, i386]
confinement: devmode
base: core18

build-packages: [intltool, pkg-config, valac]

apps:
  synapse-cli:
    command: synapse --startup
    plugs:
      - dbus
      - gpg-keys

parts:
  vala:
    plugin: nil
    override-pull: |
      apt install software-properties-common -y
      add-apt-repository ppa:vala-team/ppa -ny
      apt update


  synapse-project:
    after: [vala]
    source: https://github.com/MichaelAquilina/synapse-project
    source-type: git
    source-tag: 0.4.6
    source-depth: 1
    plugin: autotools
    configflags: ["--prefix=/opt"]
    build-packages:
      - libgtk-3-dev
      - libglib2.0-dev
      - libgee-0.8-dev
      - libpango1.0-dev
      - libjson-glib-dev
      - libkeybinder-3.0-dev
      - libnotify-dev
      - libzeitgeist-2.0-dev
      - libappindicator3-dev
      - librest-dev
```

This is the error I get:

```
result-set.vala:60.5-60.61: error: overriding method `Synapse.ResultSet.iterator' is incompatible with base method `Gee.Iterator<G> Gee.Iterable.iterator ()': Base method expected return type `Gee.Iterator<Gee.Map.Entry<Synapse.Match,int>>', but `Gee.Iterator<Gee.Map.Entry<Synapse.Match,int>?>' was provided.
    public Gee.Iterator<Gee.Map.Entry <Match, int>?> iterator ()
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Compilation failed: 1 error(s), 10 warning(s)
Makefile:804: recipe for target 'synapse-core.vala.stamp' failed
make[3]: *** [synapse-core.vala.stamp] Error 1
make[3]: Leaving directory '/root/parts/synapse-project/build/src/core'
Makefile:403: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory '/root/parts/synapse-project/build/src'
Makefile:464: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/root/parts/synapse-project/build'
Makefile:395: recipe for target 'all' failed
make: *** [all] Error 2
Failed to run 'make -j2' for 'synapse-project': Exited with code 2.
Verify that the part is using the correct parameters and try again.
```

I'm not sure why this is happening. I am able to build by running `make` outside the snap, but somehow running it from the snap gives the above error.
